accepted/0000-pure-mode.md

```diff
@@ -0,0 +1,344 @@
+
+#### Peer dependencies
+
+Peer dependencies are resolved by npm and treated as normal dependencies.
+
+For instance, the following dependency graph,
+
+```
+ foo@1.0.0  ────┬─────> bar@2.0.0 ──( peer dependency )──> baz@^1.0.0
+                │
+                └─────> baz@1.3.0
+
+```
+
+will be converted to the following:
+
+```
+ foo@1.0.0  ────┬─────> bar@2.0.0 ────> baz@1.3.0
+                │
+                └─────> baz@1.3.0
+
+```
+
+### Edge cases
+
+#### Peer dependency conflict
+
+When a peerDependency can be resolved by more than one version, virtual packages will be created, as many as they are versions resolving the peer dependency. These virtual packages will be the same, with the difference that their dependencies will be different and as a consequence their hash-based name will be different.
+
+For instance, the following dependency graph,
+
+```
+ cat@1.0.0  ────┬─────> baz@1.5.0
+                │
+                └─────>
+                        bar@2.0.0 ──( peer dependency )──> baz@^1.0.0
+                ┌─────>
+                │
+ foo@1.0.0  ────┴─────> baz@1.3.0
+
+```
+
+will be converted to the following:
+
+```
+ cat@1.0.0  ────┬─────> baz@1.5.0
+                │
+                └─────> bar@2.0.0 ────> baz@^1.5.0
+
+                ┌─────> bar@2.0.0 ────> baz@^1.3.0
+                │
+ foo@1.0.0  ────┴─────> baz@1.3.0
+
+```
+
+#### Circular dependencies
+
+Circular symlinks are allowed by the OS and _most_ tools work well with them. So this is not a major concern.
+
+Since the hash of a package contain its dependencies, circular dependencies make the calculation of this hash more complicated. Here is [an article](https://www.fugue.co/blog/2016-05-18-cryptographic-hashes-and-dependency-cycles.html) explaining how to hash a graph with cycles.
+
+### Simple example
+
+#### Dependency graph
+
+```
+  root
+   ┬
+   │
+   ├───> foo (workspace)
+   │      ┬
+   │      │
+   │      └───> A @ 1.0.0
+   │              ┬
+   │              │
+   │              └───> B @ 1.0.0
+   │
+   ├───> bar (workspace)
+   │      ┬
+   │      │
+   │      └───> A @ 1.0.0
+   │              ┬
+   │              │
+   │              └───> B @ 1.0.0
+   │
+   └───> cat (workspace)
+          ┬
+          │
+          └───> B @ 1.0.0
+
+```
+
+#### Installation on disk
+
+```
+  root /
+   ┬
+   │
+   ├─> node_modules / .npm / ─┬─> A@1.0.0-21f95f7 / node_modules / A / ─┬─> [content of package A]
+   │                          │                                         │
+   │                          │                                         └─> node_modules / B ( symlink to ../../../../B@1.0.0-0d98566/node_modules/B )
```

Comment:
This is not a good idea. This will result in circular symlinks. The symlink to be should be at 

```
├─> node_modules / .npm / ─┬─> A@1.0.0-21f95f7 / node_modules / B
```

From the docs of pnpm

> avoid circular symlinks. Dependencies of packages are placed in the same folder in which the dependent packages are. For Node.js it doesn't make a difference whether dependencies are inside the package's node_modules or in any other node_modules in the parent directories.

https://pnpm.io/symlinked-node-modules-structure

Reply:
Thanks a lot @zkochan, it makes total sense, I updated the RFC